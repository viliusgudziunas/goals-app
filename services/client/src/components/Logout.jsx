import React, { useEffect, useContext } from 'react';
import { Link } from 'react-router-dom';

import { GlobalContext } from '../context/GlobalState';

const Logout = () => {
  const { logout } = useContext(GlobalContext);
  const logoutUser = () => {
    window.localStorage.clear();
    logout();
  };

  useEffect(() => {
    logoutUser();
  }, []);

  return (
    <div className='container'>
      <br />
      <p>
        You are now logged out. Click <Link to='/login'>here</Link> to log back
        in
      </p>
    </div>
  );
};

export default Logout;
