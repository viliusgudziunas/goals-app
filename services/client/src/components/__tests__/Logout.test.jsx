import React from 'react';
import { shallow } from 'enzyme';
import { MemoryRouter as Router } from 'react-router-dom';

import Logout from '../Logout';

describe('<Logout />', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(
      <Router location='/'>
        <Logout />
      </Router>
    );
    expect(wrapper.html()).toMatchSnapshot();
  });

  const wrapper = shallow(<Logout />);
  it('should have a logged out message', () => {
    expect(wrapper.find('p').length).toBe(1);
    expect(wrapper.find('p').get(0).props.children[0]).toContain(
      'You are now logged out.'
    );
  });
});
