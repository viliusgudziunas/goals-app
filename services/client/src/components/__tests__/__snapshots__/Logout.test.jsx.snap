// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logout /> should match the snapshot 1`] = `"<div class=\\"container\\"><br/><p>You are now logged out. Click <a href=\\"/login\\">here</a> to log back in</p></div>"`;
